storage/rocksdb/ha_rocksdb.cc

```diff
@@ -8322,9 +8385,21 @@ int ha_rocksdb::update_indexes(const struct update_row_info &row_info,
                                const bool &pk_changed) {
   int rc;
 
+  // If this table has a non-hidden PK, update it first.
```

Comment:
Can you add additional comment that the pk must be updated first to pull out the ttl fields? Also, I don't think you need to special case out has_hidden_pk(). You should be able to handle them both here and let the loop handle sk only.